crates/compiler/test_gen/src/wasm_str.rs

```diff
@@ -630,6 +630,156 @@ fn str_from_utf8_fail_surrogate_half() {
     );
 }
 
+#[test]
```

Comment:
It's not great that we have all of these `gen_str` tests duplicated instead of being in a single file.

@bhansconnect from the `git blame` of the comment at the top of this file, it looks like you were the dev to initially [make this module separate for pointer width reasons](https://github.com/roc-lang/roc/commit/360974398abeb705a68ec22121db15cd0bdb308c#diff-0207dec318996ddc5cf532d5107da1d82a936ef2858c42eaac48a4834a4ed22e). Would you think it a bad idea to try to merge most of these tests into `test_gen/src/gen_str.rs` with all three targets, aka

```rust
#[cfg(any(feature = "gen-llvm", feature = "gen-dev", feature = "gen-wasm"))]
```

Reply:
I had the same thought while writing, I intended to bring it up as another PR but got lost debugging the mac issue. Maybe we can make an issue to merge these tests (assuming there's not some other reason blocking this)?

Reply:
Yeah, an issue would be totally fine. Let's get these functions merged!

Reply:
Honestly don't recall at this point. If possible, sounds great to merge